src/libs/auth.js

```diff
@@ -110,24 +131,26 @@ export const initializeAuth = _.memoize(async () => {
         fenceStatus: isSignedIn ? state.fenceStatus : {},
         // Load whether a user has input a cookie acceptance in a previous session on this system,
         // or whether they input cookie acceptance previously in this session
-        cookiesAccepted: isSignedIn ? state.cookiesAccepted || getLocalPrefForUserId(user.getId(), cookiesAcceptedKey) : undefined,
+        cookiesAccepted: isSignedIn ? state.cookiesAccepted || getLocalPrefForUserId(userId, cookiesAcceptedKey) : undefined,
         isTimeoutEnabled: isSignedIn ? state.isTimeoutEnabled : undefined,
         user: {
-          token: authResponse ? authResponse.access_token : undefined,
-          id: user.getId() || undefined,
+          token: user?.access_token || undefined,
+          id: userId || undefined,
           ...(profile ? {
-            email: profile.getEmail(),
-            name: profile.getName(),
-            givenName: profile.getGivenName(),
-            familyName: profile.getFamilyName(),
-            imageUrl: profile.getImageUrl()
+            email: profile.email,
+            name: profile.name,
+            givenName: profile.givenName,
+            familyName: profile.familyName,
+            imageUrl: profile.picture
           } : {})
         }
       }
     })
   }
-  processUser(getAuthInstance().currentUser.get())
-  getAuthInstance().currentUser.listen(processUser)
+  const userManager = new UserManager(getOidcConfig())
+  processUser(await userManager.getUser())
+  getAuthInstance().events.addUserLoaded(processUser)
+  getAuthInstance().events.addUserUnloaded(processUser)
 })
 
 // This is intended for tests to short circuit the login flow
```

Comment:
I do wonder if the code down below at line 160 should change.

Reply:
I was thinking about that. Looks like it is just introspecting a provided Google token and updating state, short circuiting the oauth flow. So it still works today given we still use Google auth. Once we fully switch to B2C we might need a different approach.